### PR TITLE
Coresets API proposal

### PR DESCRIPTION
**Pull request by Dref360**
## Summary:

New coreset API that I would like to implement.

### Features:

Related to #67

### CODE REVIEW

File: rfcs/02-coresets.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Indices = List[int]
+
+class BaseCoreset:
+    def get_ranks(features: Array, logits: Array) -> Indices:
+        # K-Means or something
+        pass
+        
+    def filter(logits, heuristic, proportion=1.)-> Indices:
+        # Use `heuristic` to rank features and return the top `proportion` indices.
+        pass
```
Comment (parmidaatg) [inline rfcs/02-coresets.md:24]: should we have a way to give more importance to coreset results or to heuristic results?

File: rfcs/02-coresets.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+**API**
+```python
+Array = np.ndarray
+Indices = List[int]
+
+class BaseCoreset:
+    def get_ranks(features: Array, logits: Array) -> Indices:
+        # K-Means or something
+        pass
```
[the remark this answers is not in the document]
Comment (Dref360) [reply rfcs/02-coresets.md:20]: In heuristics we do not have that option. 
I will add it.

File: rfcs/02-coresets.md (hunk 1 above)
Comment (Dref360) [reply rfcs/02-coresets.md:24]: Not sure how to merge them.
The idea of this method is to use BALD to create a pool of potential candidates (ie. the top 10%) and select from them with a coreset. I will remove it as we don't know if it works.

**Review by parmidaatg — APPROVED** [review]
I think it is a nice proposal. We'll see in action what other changes needed. I'll merge

**Comment by GeorgePearse** [comment]
Did this idea get anywhere in the end? e.g. a weighted balance between coreset selection and some other heuristic (e.g. BALD) ?

**Comment by GeorgePearse** [comment]
I'd be very tempted to work on it. With the view that we'd be creating a class to combine any two (or more?) strategies with some weighting.

Independently of that I think being able to intialize with coreset (and other diversity metrics) with a very simple API would be nice for beginners.

e.g. for the minute I have 

```
baal_data_module.active_set.label_randomly(hyperparams.initial_pool)
```

it would be nice to have 

```
baal_data_module.active_set.label_init(hyperparams.initial_pool, strategy='coreset')
```
